File: Algorithms/Searches/C/LinearSearch.c
#include <stdio.h>

int linearSearch(int array[], int size, int key)
{
    for (int i = 0; i < size; i++)
    {
        if (array[i] == key)
        {
            return i;
        }
    }
    return -1;
}

int main()
{
    int array[10] = {9, 1, 8, 2, 7, 10, 3, 6, 4, 5};
    int key = 10;
    int index = linearSearch(array, 10, key);
    if (index == -1)
    {
        printf("Element not found");
    }
    else
    {
        printf("Element found at index %d", index);
    }
}

File: Algorithms/Searches/Csharp/LinearSearch.cs
public static class LinearSearch
{
    
    // Implement LinearSearch as extension method of int array
    // Our method returns index of target element.
    
    public static int LinSearch(this int[] array, int target)
    {
        for (int i = 0; i < array.Length; i++)
        {
            if (array[i] == target)
            {
                return i;
            }
        }

        return -1; // If there is no target element in the array, return -1 
    }
    
    
    public static void Main()
    {
        int[] array = new int[]{ 9, 8, 7, 5, 4, 3, 2, 1};
        
        Console.WriteLine("Our array:");
        
        int target = 5;
        
        Console.WriteLine($"Looking for an element that is equal to {target}.}");
        
        Console.WriteLine($"Element found at: {array.LinSearch(target)}");
    }

    private static void PrintArray(int[] array)
    {
        for (int i = 0; i < array.Length; i++)
        {
            Console.Write($"{array[i]} ");
        }
        Console.WriteLine(";");
    }
}